[CSDiagnostics] Offer a fix-it to insert a return type placeholder when returning from a void function

theblixguy opened the pull request:
Currently, when we return from a `Void`-returning function, an error is emitted:

```swift
func voidFunc() {
  return 0 // error: unexpected non-void return value in void function
}
```

It's likely that the user forgot to add a return type, so add a fix-it to insert a placeholder for the type.

theblixguy committed:
[CSDiagnostics] Offer a fix-it to insert a return type when returning from a void function

xedin commented on lib/Sema/CSDiagnostics.cpp:
```diff
@@ -4792,6 +4792,21 @@ bool InvalidUseOfAddressOf::diagnoseAsError() {
+      auto fixItLoc = Lexer::getLocForEndOfToken(
+          getASTContext().SourceMgr, FD->getParameters()->getEndLoc());
+      emitDiagnostic(anchor->getLoc(), diag::add_return_type_note)
+          .fixItInsert(fixItLoc, " -> <# Return Type #>");
```
Let's jam it all together here so placeholder is `<#ReturnType#>` :)

theblixguy committed:
[CSDiagnostics] Make sure the function name is not empty

The function name will be empty in some cases, for example for property setters. In cases where the function name is empty, skip the note and fix-it.

theblixguy commented:
@swift-ci please smoke test

xedin commented:
@theblixguy Let's hold off from triggering CI for now, it would have to be re-triggered after LLVM changes are done anyway.

theblixguy committed:
[Test] Update existing diagnostics

theblixguy commented:
@swift-ci please smoke test
